### ISSUE TITLE
Diff editor doesn't word wrap

### ISSUE THREAD

**Issue by lishaduck** [microsoft/vscode#217386]
Type: <b>Bug</b>

Since enabling the new diff editor a year or so ago, I've noticed that it doesn't word-wrap the original side, just the new side.

Reproduce:
1. in a git repository, edit a line that's over 80 characters long, say in a .md file.
2. Open the diff from the scm view.
3. Notice that it'll wrap on the right, but not on the left.

I assume that word-wrapping would need to be enabled.

VS Code version: Code 1.90.2 (Universal) (5437499feb04f7a586f677b155b039bc2b3669eb, 2024-06-18T22:37:41.291Z)
OS version: Darwin arm64 23.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|3, 3, 4|
|Memory (System)|8.00GB (0.06GB free)|
|Process Argv|--crash-reporter-id cb9ad7f5-d968-4f8f-8dad-53d8b2f60d9c|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (87)</summary>

I use quite a few languages for school and hobbies, so I rack up many extensions.
I don't this could be caused by them, but it's in a details, so I left it.
Thanks to the person who decided to create conditional loading 😄

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.2
tsl-problem-matcher|amo|0.6.2
github-markdown-preview|bie|0.3.0
markdown-checkbox|bie|0.4.0
markdown-emoji|bie|0.3.0
markdown-footnotes|bie|0.1.1
markdown-mermaid|bie|1.23.1
markdown-preview-github-styles|bie|2.0.4
markdown-yaml-preamble|bie|0.1.0
freezed|bla|0.9.6
shell-syntax|bma|1.0.5
vscode-tailwindcss|bra|0.10.5
github-code-owners|chd|4.1.0
gitignore|cod|0.9.0
dart-code|Dar|3.90.0
flutter|Dar|3.90.0
vscode-markdownlint|Dav|0.55.0
markdown-sup|Dev|1.0.6
gitlens|eam|15.1.0
vscode-html-css|ecm|2.0.10
gdscript-formatter-linter|Edd|1.3.1
prettier-vscode|esb|10.4.0
myst-highlight|Exe|0.11.0
foam-vscode|foa|0.25.11
kotlin|fwc|0.2.34
codespaces|Git|1.16.23
copilot|Git|1.207.0
copilot-chat|Git|0.16.1
remotehub|Git|0.62.0
vscode-github-actions|git|0.26.3
vscode-pull-request-github|Git|0.90.0
mdmath|goe|2.7.4
vscode-test-explorer|hbe|2.21.1
vscode-htmlhint|HTM|1.0.5
svg|joc|1.5.3
implicit-parentheses|Jor|1.1.0
commitlint|jos|2.6.0
vscode-gutter-preview|kis|0.31.2
vscode-rdbg|Koi|0.2.2
Kotlin|mat|1.7.1
workspace-storage-cleanup|meh|1.1.5
vscode-docker|ms-|1.29.1
vscode-devskim|MS-|1.0.33
vscode-dotnet-runtime|ms-|2.0.6
vscode-edge-devtools|ms-|2.1.5
jupyter|ms-|2024.5.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.18
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.369.0
remote-ssh|ms-|0.112.0
remote-ssh-edit|ms-|0.86.0
vscode-remote-extensionpack|ms-|0.25.0
live-server|ms-|0.4.13
remote-explorer|ms-|0.4.3
remote-repositories|ms-|0.40.0
remote-server|ms-|1.5.2
test-adapter-converter|ms-|0.1.9
vscode-github-issue-notebooks|ms-|0.0.130
vsliveshare|ms-|1.0.5932
bun-vscode|ove|0.0.12
material-icon-theme|PKi|5.4.2
gdscript-toolkit-formatter|Raz|1.2.3
vscode-xml|red|0.27.1
vscode-yaml|red|1.15.0
flutter-riverpod-snippets|rob|1.2.2
lit-plugin|run|1.4.3
lcov-lang|Ser|1.0.1
ruby-extensions-pack|Sho|0.1.9
ruby-lsp|Sho|0.7.5
html5-boilerplate|sid|1.1.1
vscode-native-svg-preview|snd|1.72.0
vscode-socket-security|Soc|1.1.0
sorbet-vscode-extension|sor|0.3.35
swift-lang|ssw|1.10.3
code-spell-checker|str|3.0.1
code-spell-checker-spanish|str|2.3.4
even-better-toml|tam|0.19.2
graphviz-interactive-preview|tin|0.3.5
pdf|tom|1.2.2
luna-paint|Tyr|0.16.0
vscode-mdx|uni|1.8.9
vscode-lldb|vad|1.10.0
turbo-vsc|Ver|2.0.0
vscode-wakatime|Wak|24.6.0
markdown-all-in-one|yzh|3.6.2

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vswsl492cf:30256860
vscorecescf:30445987
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
welcomedialog:30910333
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
accentitlementst:30995554
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
chatpanelc:31048052
dsvsc021:30996838
g316j359:31013175
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
newcmakeconfigv2:31071590
legacy_priority:31077528

```

</details>

**Comment by lishaduck** [comment]
Oh, and it doesn't happen in the merge conflict editor.

**Comment by lishaduck** [comment]
Huh. It only happens sometimes, it looks like:
<img width="826" alt="Screenshot 2024-06-25 at 10 04 31 AM" src="https://github.com/microsoft/vscode/assets/88557639/46ca6c88-5742-4ed5-af32-46bd6b16ffc6">

Both in a staged and unstaged package-lock.json. Haven't changed anything.

**Comment by andriyDev** [comment]
This is especially rough for the three-way merge conflict editor: the conflicts will be completely misaligned between the left, middle, and right areas, sometimes resulting in conflicts in the middle (working directory) being completely offscreen, making it unclear what conflict is even being referred to. Turning off line wrapping makes it behave rationally where I can see which conflicts are being represented in all three versions.

**Comment by lishaduck** [comment]
> I'm not sure that it's identical to the issue reported here, but there appears to be an asymmetry in the behavior when resizing editors.

Yeah, I've toyed around with reproing it and resizing always triggers it. It's not the only time it does (I think), but it's the major one.

**Comment by olfMombach** [comment]
This does still happen btw.
```
Version: 1.101.2
Commit: 2901c5ac6db8a986a5666c3af51ff804d05af0d4
Date: 2025-06-24T20:27:15.391Z
Electron: 35.5.1
ElectronBuildId: 11727614
Chromium: 134.0.6998.205
Node.js: 22.15.1
V8: 13.4.114.21-electron.0
OS: Linux x64 6.8.0-63-generic
```
